test/end-to-end/company.js

```diff
@@ -227,6 +227,142 @@ describe("Company endpoint", () => {
+                test_company_2 = await Company.create(test_company_data);
+
+                await Account.deleteMany({});
+
+                await Account.create({
+                    email: test_user_admin.email,
+                    password: await hash(test_user_admin.password),
+                    isAdmin: true
+                });
+            });
+
+            afterEach(async () => {
+                await test_agent
+                    .delete("/auth/login")
+                    .expect(HTTPStatus.OK);
+            });
+
+            test("should fail if not a valid id", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const res = await test_agent
+                    .get("/company/123/profile")
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.OBJECT_ID);
+            });
+
+            test("should fail if company does not exist", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .get(`/company/${id}/profile`)
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_NOT_FOUND(id));
+            });
+
+            test("should return if the offers array is empty", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_2.id}/profile`);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers).toHaveProperty("length", 0);
+            });
+
+            test("should return offers lower than X (size: 3)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const offers_2 = await Promise.all(Array(3).fill(null).map((_) =>
+                    Offer.create({
+                        ...generateTestOffer({
```

Comment:
I believe you can put all the properties inside `generateTestOffer` and it'll generate the object for you, so you don't need to wrap it and use `...`